test/test_service.py

```diff
@@ -0,0 +1,214 @@
+import os
+from subprocess import call
+import json
+import pytest
+import shutil
+
+# Getting absolute paths, names and regexes
+TEST_DIR = os.path.dirname(os.path.abspath(__file__))
+ROOT_DIR = os.path.dirname(TEST_DIR)
+SERVICE_CONFIG_NAME = "service_manifest.yml"
+SERVICE_CONFIG_PATH = os.path.join(ROOT_DIR, SERVICE_CONFIG_NAME)
+TEMP_SERVICE_CONFIG_PATH = os.path.join("/tmp", SERVICE_CONFIG_NAME)
+
+# Samples that we will be sending to the service
+sample1 = dict(
+    sid=1,
+    metadata={},
+    service_name='apkaye',
+    service_config={},
+    fileinfo=dict(
+        magic='ASCII text, with no line terminators',
+        md5='1f09ecbd362fa0dfff88d4788e6f5df0',
+        mime='text/plain',
+        sha1='a649bf201cde05724e48f2d397a615b201be34fb',
+        sha256='dadc624d4454e10293dbd1b701b9ee9f99ef83b4cd07b695111d37eb95abcff8',
+        size=19,
+        type='unknown',
+    ),
+    filename='dadc624d4454e10293dbd1b701b9ee9f99ef83b4cd07b695111d37eb95abcff8',
+    min_classification='TLP:WHITE',
+    max_files=501,  # TODO: get the actual value
+    ttl=3600,
+)
+
+
+@pytest.fixture
+def class_instance():
+    temp_service_config_path = os.path.join("/tmp", SERVICE_CONFIG_NAME)
+    try:
+        # Placing the service_manifest.yml in the tmp directory
+        shutil.copyfile(SERVICE_CONFIG_PATH, temp_service_config_path)
+
+        from apkaye.apkaye import APKaye
+        yield APKaye()
+    finally:
+        # Delete the service_manifest.yml
+        os.remove(temp_service_config_path)
+
+
+class TestAPKaye:
+
+    @classmethod
+    def setup_class(cls):
+        # Dockerfile stuff
+        call(["sudo", "apt-get", "install", "-y", "openjdk-8-jre-headless", "java-common", "libc6-i386", "lib32z1", "lib32gcc1", "unzip", "wget"])
```

Comment:
Given that someone working on a service should be able to run the coverage tests on their normal dev system, this seems a bit intrusive to include in the tests themselves.

Reply:
Addressed here: https://github.com/CybercentreCanada/assemblyline-service-apkaye/pull/1/commits/77347c0a1aa4fd4cef795bcc4734ad64919e5cdd